github/actions_secrets.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+type PublicKey struct {
+	ID  *string `json:"key_id,omitempty"`
+	Key *string `json:"key,omitempty"`
+}
+
+// GetPublicKey gets a public key that should be used for secret encryption.
+//
+// GitHub API docs: https://developer.github.com/v3/actions/secrets/#get-your-public-key
+func (s *ActionsService) GetPublicKey(ctx context.Context, owner string, repo string) (*PublicKey, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/secrets/public-key", owner, repo)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pubKey := new(PublicKey)
+	resp, err := s.client.Do(ctx, req, pubKey)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pubKey, resp, nil
+}
+
+// Secret represents a repository action secret.
+type Secret struct {
+	Name      *string    `json:"name,omitempty"`
+	CreatedAt *Timestamp `json:"created_at,omitempty"`
+	UpdatedAt *Timestamp `json:"updated_at,omitempty"`
+}
+
+type secrets struct {
```

Comment:
According to the docs, the full response looks like this:

```
[
  {
    "total_count": 2,
    "secrets": [
      {
        "name": "GH_TOKEN",
        "created_at": "2019-08-10T14:59:22Z",
        "updated_at": "2020-01-10T14:59:22Z"
      },
      {
        "name": "GIST_ID",
        "created_at": "2020-01-10T10:59:22Z",
        "updated_at": "2020-01-11T11:59:22Z"
      }
    ]
  }
]
```

So let's please change this struct to be:

```
// Secrets represents one item from the ListSecrets response.
type Secrets struct {
  TotalCount int `json:"total_count"`
  Secrets []*Secret `json:"secrets"`
}
```

and then we will return a slice of Secrets (`[]*Secrets`) below in `ListSecrets` to match the GitHub API docs.

Reply:
I think the API docs are wrong, when I make a request to list secrets I get an object, not an array, so I decided to return Secrets object, not a slice. I've reached out to GitHub support also regarding this API/doc discrepancy, more details below.